File: Python/Ch. 07. R - Binomial Distribution - Probability of x between x1 and x2.py
# R - BINOMIAL DISTRIBUTION - PROBABILITY of x BETWEEN x1 and x2
# http://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.binom.html

from scipy.stats import binom

# Enter the following values
n = 10          # number of trials
p = 0.2         # probability of success 
x1 = 3          # minimum number of successes
x2 = 7          # maximum number of successes

print binom.cdf(x2, n, p) - binom.cdf(x1, n, p) +  binom.pmf(x1, n, p)

File: R/Ch. 07. R - Binomial Distribution - Probability of x between x1 and x2.R
# R - BINOMIAL DISTRIBUTION - PROBABILITY of x BETWEEN x1 and x2

# Enter the following values:

number.of.trials = 10
probability.of.success = 0.2
minimum.number.of.successes = 3
maximum.number.of.successes = 7

# Evaluation

n = number.of.trials
p = probability.of.success
x1 = minimum.number.of.successes
x2 = maximum.number.of.successes
P = pbinom(x2, n, p)-pbinom(x1, n, p)+dbinom(x1, n, p)

# Probability P(x) that the number of successes lies between x1 and x2 (both included):      

P

